src/configs.ts

```diff
@@ -104,6 +104,41 @@ export async function readConfig(
   if (!Array.isArray(rawConfigs))
     rawConfigs = [rawConfigs]
 
+  // ESLint applies these default configs to all files
+  // https://github.com/eslint/eslint/blob/21d3766c3f4efd981d3cc294c2c82c8014815e6e/lib/config/default-config.js#L66-L69
+  rawConfigs.unshift(
+    {
+      name: 'eslint/defaults/languages',
+      languageOptions: {
+        sourceType: 'module',
+        ecmaVersion: 'latest',
+        parserOptions: {},
+      },
+      linterOptions: {
+        reportUnusedDisableDirectives: 1,
+      },
+    } as FlatConfigItem,
+    {
+      name: 'eslint/defaults/ignores',
+      ignores: [
+        '**/node_modules/',
+        '.git/',
+      ],
+    } as FlatConfigItem,
```

Comment:
I don't mind either way - in the end this globbing things is more or less a approximation for reference. Would be nice if we could reuse the ESLint's logic to get the same result. Otherwise I see no much differences, and honestly would slightly prefer to keep it as-is to make sure we don't over scan.

Reply:
@antfu Added a request to expose the ESLint logic: https://github.com/eslint/eslint/issues/18619